.github/workflows/publish.yml

```diff
@@ -19,14 +19,30 @@ jobs:
 
     steps:
     - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
     - name: Set up Python
       uses: actions/setup-python@v4
       with:
         python-version: "3.11"
+        
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
         pip install build
+
+    - name: Check git status
+      run: |
+        git status
+        git log -n 2
+        git diff
+
```

Comment:
What's this for? I'm imagining the intention may be to abort if there are local changes, but I don't think this even does that?

Reply:
Yes, it just prints things for debugging. Any local changes change the tag and break a release on PyPI. I have found it handy to fix the current issue, and don't see much reason to drop it, as if there is a failure in future we will have more info, and it doesn't exactly take any time to execute.